bindings/python/FloatingBaseEstimators/tests/test_legged_odometry.py

```diff
@@ -11,19 +11,55 @@ def test_legged_odometry():
     # create KinDynComputationsDescriptor
     kindyn_handler = blf.parameters_handler.StdParametersHandler()
     kindyn_handler.set_parameter_string("model_file_name", "./model.urdf")
-    kindyn_handler.set_parameter_vector_string("joints_list", ["neck_pitch", "neck_roll", "neck_yaw",
-                                                               "torso_pitch", "torso_roll", "torso_yaw",
-                                                               "l_shoulder_pitch", "l_shoulder_roll", "l_shoulder_yaw",
-                                                               "l_elbow",
-                                                               "r_shoulder_pitch", "r_shoulder_roll", "r_shoulder_yaw",
-                                                               "r_elbow",
-                                                               "l_hip_pitch", "l_hip_roll", "l_hip_yaw",
-                                                               "l_knee", "l_ankle_pitch", "l_ankle_roll",
-                                                               "r_hip_pitch", "r_hip_roll", "r_hip_yaw",
-                                                               "r_knee", "r_ankle_pitch", "r_ankle_roll"])
+    joints_list = ["neck_pitch", "neck_roll", "neck_yaw",
+                   "torso_pitch", "torso_roll", "torso_yaw",
+                   "l_shoulder_pitch", "l_shoulder_roll", "l_shoulder_yaw","l_elbow",
+                   "r_shoulder_pitch", "r_shoulder_roll", "r_shoulder_yaw","r_elbow",
+                   "l_hip_pitch", "l_hip_roll", "l_hip_yaw","l_knee", "l_ankle_pitch", "l_ankle_roll",
+                   "r_hip_pitch", "r_hip_roll", "r_hip_yaw","r_knee", "r_ankle_pitch", "r_ankle_roll"]
+    kindyn_handler.set_parameter_vector_string("joints_list", joints_list)
     kindyn_desc = blf.floating_base_estimators.construct_kindyncomputations_descriptor(kindyn_handler)
     assert kindyn_desc.is_valid()
 
+    # Check the number of degrees of freedom of the model
+    assert kindyn_desc.get_nr_of_dofs() == len(joints_list)
+
+    # Set the joint positions to random values
+    joint_values = [np.random.uniform(-0.5,0.5) for _ in range(kindyn_desc.get_nr_of_dofs())]
+    assert kindyn_desc.set_joint_pos(joint_values)
+
+    # Check that the joint positions are all set to the desired values
+    updated_joint_pos = np.zeros(kindyn_desc.get_nr_of_dofs())
+    assert kindyn_desc.get_joint_pos(updated_joint_pos)
+    assert updated_joint_pos == pytest.approx(joint_values)
+
+    # Get the robot state
+    # TODO: world_T_base.flags['F_CONTIGUOUS'] is required to be True
+    world_T_base = np.asfortranarray(np.zeros((4,4)))
```

Comment:
This is weird, I'll leave this comment as placeholder for the final review stage. At first look I though that pybind already performs some operation between numpy as eigen, I'm not sure though if the serialization major (column or row) is automatically detected.

Reply:
I found it weird as well. When I originally performed the pytests without this weird fix they were failing. In the current version of the PR instead the fix is not required and I cannot reproduce the error. 

I will simply replace `np.asfortranarray()` with the usual `np.asarray()`.

Reply:
Mmh ok, just to understand, did you understand what changed wrt to the initial status of the PR? The fortran vs default arrays regards the row-major vs column major serialization (i.e. how a matrix is flattened to an array).

Reply:
Not really, but something with the pytest was weird from the very beginning since in the initial status of the PR the pytests succeeded in my local configuration while they failed here (see https://github.com/dic-iit/bipedal-locomotion-framework/pull/303#issuecomment-836792318).